storage/storage_image.go

```diff
@@ -1011,6 +1011,18 @@ func (s *storageImageDestination) Commit(ctx context.Context, unparsedToplevel t
 	} else {
 		logrus.Debugf("created new image ID %q", img.ID)
 	}
+
+	// Clean up the unfinished image on any error.
+	// (Is this the right thing to do if the image has existed before?)
+	commitSucceeded := false
+	defer func() {
+		if !commitSucceeded {
```

Comment:
I feel quite tempted to do a `logrus.Errorf()` here.

If the image couldn't be committed, leaving an explicit and obvious breadcrumb to the user may save some debugging time.

Reply:
Yes, a failure to delete an image suggests storage problems that should be reported to the user.

I have changed this as a separate commit, to differentiate from the cleanup.

Reply:
Sorry, I was vague before.  I would welcome a log indicating that the commit has failed and that the image is going to be deleted from the storage.

Reply:
If the image existed before an unsuccessful pull, some users may worry why it has been deleted.  Tracking that down can be time costly.

Reply:
Sure, added a log line in that case, including the old names that are now untagged. (I haven’t tested this manually, though.)